ledger/ledger-api-auth/src/test/suite/scala/com/digitalasset/ledger/api/auth/AuthServiceJWTCodecSpec.scala

```diff
@@ -210,6 +226,75 @@ class AuthServiceJWTCodecSpec
         parse(serialized) shouldBe Success(expected)
       }
 
+      "support additional daml user token with prefixed audience" in {
+        val serialized =
+          """{
+            |  "aud": "https://daml.com/jwt/aud/participant/someParticipantId",
+            |  "sub": "someUserId",
+            |  "exp": 100
+            |}
+          """.stripMargin
+        val expected = StandardJWTPayload(
+          participantId = Some("someParticipantId"),
+          userId = "someUserId",
+          exp = Some(Instant.ofEpochSecond(100)),
+          format = StandardJWTTokenFormat.ParticipantId,
+        )
+        parse(serialized) shouldBe Success(expected)
+      }
+
+      "treat a singleton array of audiences equivalent to a string of its first element" in {
+        val prefixed =
+          """{
+            |  "aud": ["https://daml.com/jwt/aud/participant/someParticipantId"],
+            |  "sub": "someUserId",
+            |  "exp": 100
+            |}
+          """.stripMargin
+        parse(prefixed) shouldBe Success(
+          StandardJWTPayload(
+            participantId = Some("someParticipantId"),
+            userId = "someUserId",
+            exp = Some(Instant.ofEpochSecond(100)),
+            format = StandardJWTTokenFormat.ParticipantId,
+          )
+        )
+
+        val standard =
+          """{
+            |  "aud": ["someParticipantId"],
+            |  "sub": "someUserId",
+            |  "exp": 100,
+            |  "scope": "dummy-scope1 daml_ledger_api dummy-scope2"
+            |}
+          """.stripMargin
+        parse(standard) shouldBe Success(
+          StandardJWTPayload(
+            participantId = Some("someParticipantId"),
+            userId = "someUserId",
+            exp = Some(Instant.ofEpochSecond(100)),
+            format = StandardJWTTokenFormat.Scope,
+          )
+        )
+      }
+
+      "support additional daml user token with prefixed audience and provided scope" in {
```

Comment:
I’d suggest to add two more tests:
1. Token with empty participant id but the prefix. I wonder if that should actually get rejected outright instead of being interpreted as a different token. I don’t really see any point in trying to interpret it as a different token.
2. A token with multiple audiences